File: src/main/java/com/typosafe/java/Pojo.java
package com.typosafe.java;

public class Pojo {
  private String name = "Ala";
  private int age = 1;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public int getAge() {
    return age;
  }

  public void setAge(int age) {
    this.age = age;
  }

  public static String updateFromLoginStatic(Pojo pojo, String login){
    pojo.setName(login);
    pojo.setAge(login.length());
    return pojo.getName() + "-" + pojo.getAge();
  }
}

File: src/main/kotlin/com/typosafe/kotlin/Pojo.kt
package com.typosafe.kotlin


class Pojo {
    var name: String = "ala"
    var age: Int = 123

    companion object {
        fun updateFromLoginCompanion(pojo: Pojo, login: String): String {
            pojo.name = login
            pojo.age = login.length
            return "${pojo.name}-${pojo.age}"
        }
    }
}

fun Pojo.login(): String = "$name-$age"